exercises/concept/meltdown-mitigation/conditionals_test.py

```diff
@@ -7,164 +7,70 @@
 
 
 class TestConditionals(unittest.TestCase):
-    # Checking the first condition using assertTrue and assertFalse
-    # The values for arguments is not final and should be considered as placeholders
-    # More test-cases  required for full testing
+    # Test cases for is_criticality_balanced()
 
+    # Testing border cases around typical points T, n == (800, 500), (625, 800), (500, 1000)
     @pytest.mark.task(taskno=1)
-    def test_is_criticality_balanced_set1(self):
-        self.assertTrue(is_criticality_balanced(
-            temperature=750, neutrons_emitted=650), msg="Expected True but returned False")
+    def test_is_criticality_balanced(self):
+        for temperature, neutrons_emitted, expected in (
+            (750, 650, True), (799, 501, True), (500, 600, True), (1000, 800, False),
+            (800, 500, False), (800, 500.01, False), (799.99, 500, False), (500.01, 999.99, False),
+            (625, 800, False), (625.99, 800, False), (625.01, 799.99, False),
+            (799.99, 500.01, True), (624.99, 799.99, True),
+            (500, 1000, False), (500.01, 1000, False), (499.99, 1000, True),
+            ):
+            got = is_criticality_balanced(temperature=temperature, neutrons_emitted=neutrons_emitted)
+            self.assertEqual(
+                got,
+                expected,
+                msg=f"Expected {expected} but returned {got} with T={temperature} and neutrinos={neutrons_emitted}"
+            )
```

Comment:
I love that you parameterized these, but would prefer that you use `unittest.subtest` as opposed to a plain loop.  

That way, individual points can count as individual tests when they fail, and we can count/label each as a variant.  See [this exercise](https://github.com/exercism/python/blob/main/exercises/concept/guidos-gorgeous-lasagna/lasagna_test.py) as an example. And a screengrab of a failure state in the UI with the variants listed:
![2021-09-10_05-35-43](https://user-images.githubusercontent.com/5923094/132854137-acf564d8-9327-472e-9d6d-4403b2d0b7bd.png)

Reply:
Yes -- that peculiarity you mentioned is .. really a major pain for subtests.  😅 

For the runner/UI, we apprehend the "success" and re-write the parent outcome as a failure if any `subtest` is a failure.  But if someone is running Pytest locally, it can be confusing -- even with `pytest-subtest`.  

Still thinking on how to clarify that for CLI students without packaging up our own PyTest plugin for download and distribution (_which might be an option_).  But I am also of the opinion that anyone who has figured out the CLI will only be stopped by the subtest mis-match once or twice.  And I thought having shorter, cleaner, and clearer test files was worth it.

Reply:
Yes, I got quickly used to the funny output. 
It's just annoying that running `pytest` on a default empty stub (where all functions just `pass`), one gets: 
```
=== 41 failed, 3 passed in 0.76s ====
```
3 passed tests before even writing a single chracter of code. 🙄

Reply:
I know!  It makes me _insane_.  Evidently, it's not an "easy" problem to fix?  Once I saw this ongoing [debate/discussion set](https://github.com/pytest-dev/pytest-subtests/issues/11), I .... _backed away slowly_.  LOL

Reply:
This next review will take me a few - I need to run the new test file through the runner and make sure the failures get reported correctly.  I _may_ need to change `point` to `variant`, but I need to double check the runner code first.  😄